### PR TITLE
Move mpp-interconnect from pulse to CCP

### PR DESCRIPTION
**Pull request by pengzhout**

### CODE REVIEW

**Comment by pengzhout** [comment]
@kmacoskey thanks a lot for your suggestion, can you help to review this PR?

File: concourse/tasks/run_tinc.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ run:
     set -ex
     ccp_src/aws/setup_ssh_to_cluster.sh
     if [ -n "$PRE_TEST_SCRIPT" ]; then
-      ssh -t mdw "$CUSTOM_ENV $PRE_TEST_SCRIPT"
+      if [ -n "$PRE_TEST_SCRIPT_USER" ]; then
+        ssh -t "$PRE_TEST_SCRIPT_USER"@mdw "$CUSTOM_ENV $PRE_TEST_SCRIPT"
+      else
+        ssh -t mdw "$CUSTOM_ENV $PRE_TEST_SCRIPT"
+      fi
```
Comment (pivotal-ning-yu) [inline concourse/tasks/run_tinc.yml]: Maybe the above fine lines can be replaced with one single line:

```bash
ssh -t "${PRE_TEST_SCRIPT_USER:-$USER}"@mdw "$CUSTOM_ENV $PRE_TEST_SCRIPT"
```

**Comment by hlinnaka** [comment]
Is the kernel module distributed by the same Apache 2 license as the rest of the Greenplum codebase? Is that OK, given that the Linux kernel itself is GPLv2?

File: src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+{
+	struct connectionInfo *iter = NULL;
+	int			hashcode = 0;
+
+	if (header == NULL)
+	{
+		return NULL;
+	}
+
+	hashcode = CONN_HASH_VALUE(header) % (ht->size);
```
Comment (paul-guo-) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c]: I'd suggest changing hashcode (there are more than 1 places) to unsigned type. I encountered a panic due to this (ht->table[hashcode] is illegal finally).

By the way, some file headers are missing.

**Comment by kmacoskey** [comment]
> Is the kernel module distributed by the same Apache 2 license as the rest of the Greenplum codebase? Is that OK, given that the Linux kernel itself is GPLv2?
@pivotal-mike

**Comment by pivotal-mike** [comment]
@hlinnaka It should be OK to distribute as it.

@pengzhout can we please verify the copyright on these source files.  They aren't correct for Pivotal and I want to ensure we keep the correct providence of the code.  I'm going to ping you off line about that just to be sure.

**Comment by pivotal-ning-yu** [comment]
Linux kernel modules can have their own licenses, here is a brief
description on wiki:
https://en.wikipedia.org/wiki/Loadable_kernel_module#License_issues

One example is the nvidia video linux drivers.

**Comment by pengzhout** [comment]
@pivotal-mike we wrote those code years ago, so it's ok to use it, I will add pivotal copyright to each file.

**Comment by pengzhout** [comment]
updated the copyright headers.

File: src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c (hunk 2 above)
Comment (pengzhout) [reply src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c]: fixed, thanks.

File: concourse/tasks/run_tinc.yml (hunk 1 above)
Comment (pengzhout) [reply concourse/tasks/run_tinc.yml]: fixed, thanks.

File: src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c [hunk 3]
```diff
@@ -0,0 +1,231 @@
+/*-------------------------------------------------------------------------
+ *
+ * connectionTable.c
+ *	  Functions to manage virtual interconnect connections.
+ *
+ * Portions Copyright (c) 2012-Present Pivotal Software, Inc.
+ *
```
Comment (yydzero) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c]: Remove 'Portions'

File: src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/*-------------------------------------------------------------------------
+ *
+ * connectionTable.h
+ *
+ * Portions Copyright (c) 2012-Present Pivotal Software, Inc.
+ *
```
Comment (yydzero) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.h]: Remove Portions for all files written by Pivotal

File: src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c (hunk 3 above)
Comment (pengzhout) [reply src/test/tinc/tincrepo/mpp/gpdb/tests/dispatch/interconnect/ickm/connectionTable.c]: Fixed, thanks

**Pull request merged**